app/models/course/condition/assessment.rb

```diff
@@ -83,6 +84,11 @@ def validate_unique_dependency
     errors.add(:assessment, :unique_dependency)
   end
 
+  def validate_acyclic_dependency
+    return unless cyclic?
+    errors.add(:assessment, :cyclic_dependency)
```

Comment:
Wanted to but there might be multiple path that result in the cyclic. Not really sure how to present it to make it meaningful for the user. So decided to keep it simple and only let the user know that this condition leads to a cyclic dependency.